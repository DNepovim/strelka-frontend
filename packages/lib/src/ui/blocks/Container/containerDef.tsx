import { v4 as uuid } from "uuid"
import { BlockTemplates, InputType } from "../enums"
import { Block, BlockDef } from "../blocks"
import { BlockFields } from "../../components/Block/Block"
import ProfileOutlined from "@ant-design/icons/lib/icons/ProfileOutlined"

export interface ContainerBlock extends Block {
  template: BlockTemplates.Container
  fields: ContainerProps
}

export interface ContainerProps extends BlockFields<{ id: string }[]> {}

export const containerDef: BlockDef<ContainerProps> = {
  title: "Blok",
  template: BlockTemplates.Container,
  inputs: InputType.Container,
  getDefautlValues: () => ({
    content: [
      {
        id: uuid(),
        template: BlockTemplates.Heading,
        fields: { content: "" },
      },
    ],
  }),
  icon: ProfileOutlined,
}
