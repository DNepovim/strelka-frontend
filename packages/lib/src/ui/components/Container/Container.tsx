import styled from "@emotion/styled"
import React from "react"
import { theme } from "../../styles/theme"

export const Container: React.FC = styled("div")`
  position: relative;
  z-index: 10;
  box-sizing: border-box;
  max-width: ${theme.layout.width}px;
  margin: 0 auto;
`
