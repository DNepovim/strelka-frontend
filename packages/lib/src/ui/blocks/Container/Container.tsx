import { ContainerProps } from "./containerDef"

export const Container: React.FC<ContainerProps> = ({ content }) => (
  <div>{content}</div>
)
