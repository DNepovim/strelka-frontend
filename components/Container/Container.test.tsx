import renderer from "react-test-renderer"
import { Container } from "./Container"

describe("Container component", () => {
  it("should renders correctly", () => {
    const tree = renderer
      .create(
        <Container>
          <h1>Title</h1>
        </Container>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
