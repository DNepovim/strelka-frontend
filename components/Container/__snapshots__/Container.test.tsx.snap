// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container component should renders correctly 1`] = `
.emotion-0 {
  position: relative;
  z-index: 10;
  box-sizing: border-box;
  max-width: 1000px;
  margin: 0 auto;
  padding: 100px 16px;
}

<div
  className="emotion-0"
>
  <h1>
    Title
  </h1>
</div>
`;
